Codecademy/gamut
# style(Tag): Update Tag colors

### Overview
Updates tags to use the correct color tokens.
Current implementation of tags somewhat uses ColorMode and needs to invert the colors to work. 

Major Changes:
1. removed DismissButton from TagWrapper
2. TagWrapper => TagLabelWrapper 
3. ^ necessary to remove the background behind DimissButton 
4. Tag is no longer a Background, is now a Box 
5. Changed the BaseButton to an IconButton (to gain access to ToolTip properties) 
6. The gray variant should have updated colors (and for hover state too)


Updates tags to use the correct color tokens.


### PR Checklist

- [ ] Related to designs: https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=46539-40954&m=dev 
- [x] Related to JIRA ticket: [GM-778]

#### Testing Instructions



1. Go to the Storybook preview and the tags page `?path=/docs/atoms-tag--tag-variants` 
2. Look over the tags implementation, compare against the [figma](https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=46539-40954&m=dev) 
—
Mono
1. Go to the tengu preview env
2. Go to Pricing > Individuals > select "Try Pro for free" 
3. Create an account
4. At checkout, click on the "Have a promo code?" link
5. Add in `test50` 
6. Check that the tag renders correctly. 
— 
Monolith
1. go to the preview (without portal app), log in as an admin
2. Go to the admin page and create a test user that has a Pro subscription + billed monthly (you can also use this link: 
https://pr-38096-monolith.dev-eks.codecademy.com/admin/test_user?prefill_id=66ab9a074e03aa000159fe35)
^ the settings is important because we need to upgrade the user (and can't seem to upgrade free or pro-lite users) 
3. Impersonate the test user 
4. go to "Account + Billing" > "Payment And Plan" tab 
5. Select "Change Plan" and opt to upgrade to bill yearly 
6. in the address bar append the query string: `?coupon=test50` 
7. see that the tag is rendered 
8. ... 
9. Profit 

Screenshots:
Mono:
![image](https://github.com/user-attachments/assets/83fac1ae-c189-4727-a992-c7d491f91a99)

Monolith: 
![image](https://github.com/user-attachments/assets/82b9d9ed-c314-49ef-afdc-c9ef65ae0e91)


 

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/38096)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/7067)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |




[GM-778]: https://codecademy.atlassian.net/browse/GM-778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
